### PR TITLE
[Keyboard] Idobao Montex V2(RGB)

### PR DESCRIPTION
**Pull request by peepeetee**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/idobao/montex/v2/config.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x6964  // "id"
+#define PRODUCT_ID      0x6060
+#define DEVICE_VER      0x0002
```
Comment (sigprof) [inline keyboards/idobao/montex/v2/config.h:11]: VIA issues may require changing the `PRODUCT_ID` too (although the matrix layout seems to be exactly the same as on V1, so the basic VIA JSON for V1 will still work for V2, changing from `RGBLIGHT_ENABLE` to `RGB_MATRIX_ENABLE` will require a separate VIA JSON if you want to have proper effect names there; also the backlight feature was in V1, but not in V2).

**Review by vinorodrigues — APPROVED** [review]
This variant of the Montex has been tested with the Drop version of the Montex keyboard - https://drop.com/buy/idobao-montex-numpad-hotswappable-kit .

File: keyboards/idobao/montex/v1rgb/info.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{
+    "keyboard_name": "IDOBAO MONTEX RGB",
+    "layouts": {
+        "LAYOUT": {
```
Comment (drashna) [inline keyboards/idobao/montex/v1rgb/info.json:4]:
```suggestion
        "LAYOUT_numpad_6x5": {
```

File: keyboards/idobao/montex/v1rgb/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+     * │Gui│ 7 │ 8 │ 9 │   │
+     * ├───┼───┼───┼───┤ + │
+     * │Alt│ 4 │ 5 │ 6 │   │
+     * ├───┼───┼───┼───┼───┤
+     * │Shi│ 1 │ 2 │ 3 │   │
+     * ├───┼───┴───┼───┤Ent│
+     * │MO1│ 0     │ . │   │
+     * └───┴───────┴───┴───┘
+     */
+    [0] = LAYOUT(
```
Comment (drashna) [inline keyboards/idobao/montex/v1rgb/keymaps/default/keymap.c:22]:
```suggestion
    [0] = LAYOUT_numpad_6x5(
```

File: keyboards/idobao/montex/v1rgb/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+     * │Hu-│Hom│ ↑ │PgU│   │
+     * ├───┼───┼───┼───┤Br+│
+     * │St+│ ← │   │ → │   │
+     * ├───┼───┼───┼───┤───┤
+     * │St-│End│ ↓ │PgD│   │
+     * ├───┼───┴───┼───┤Ent│
+     * │mo1│Insert │Del│   │
+     * └───┴───────┴───┘───┘
+     */
+    [1] = LAYOUT(
```
Comment (drashna) [inline keyboards/idobao/montex/v1rgb/keymaps/default/keymap.c:46]:
```suggestion
    [1] = LAYOUT_numpad_6x5(
```

File: keyboards/idobao/montex/v1rgb/keymaps/via/config.h [hunk 5]
```diff
@@ -0,0 +1,8 @@
+// Copyright 2022 peepeetee (@peepeetee)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#if defined(RGB_MATRIX_ENABLE)
+    #define VIA_QMK_RGBLIGHT_ENABLE
+#endif
```
Comment (drashna) [inline keyboards/idobao/montex/v1rgb/keymaps/via/config.h:8]: Won't be needed on develop.

File: keyboards/idobao/montex/v1rgb/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+     * │Hu-│Hom│ ↑ │PgU│   │
+     * ├───┼───┼───┼───┤Br+│
+     * │St+│ ← │   │ → │   │
+     * ├───┼───┼───┼───┤───┤
+     * │St-│End│ ↓ │PgD│   │
+     * ├───┼───┴───┼───┤Ent│
+     * │mo1│Insert │Del│   │
+     * └───┴───────┴───┘───┘
+     */
+    [_FN] = LAYOUT(
```
Comment (drashna) [inline keyboards/idobao/montex/v1rgb/keymaps/via/keymap.c:53]:
```suggestion
    [_FN] = LAYOUT_numpad_6x5(
```

File: keyboards/idobao/montex/v1rgb/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+     * │   │   │   │   │   │
+     * ├───┼───┼───┼───┤   │
+     * │   │   │   │   │   │
+     * ├───┼───┼───┼───┤───┤
+     * │   │   │   │   │   │
+     * ├───┼───┴───┼───┤   │
+     * │   │       │   │   │
+     * └───┴───────┴───┘───┘
+     */
+    [_EMPTY2] = LAYOUT(
```
Comment (drashna) [inline keyboards/idobao/montex/v1rgb/keymaps/via/keymap.c:77]:
```suggestion
    [_EMPTY2] = LAYOUT_numpad_6x5(
```

File: keyboards/idobao/montex/v1rgb/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+        _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______,
+        _______, _______, _______, _______,
+        _______, _______, _______, _______, _______,
+        _______, _______,           _______
+    ),
+
+    [_EMPTY3] = LAYOUT(
```
Comment (drashna) [inline keyboards/idobao/montex/v1rgb/keymaps/via/keymap.c:86]:
```suggestion
    [_EMPTY3] = LAYOUT_numpad_6x5(
```

File: keyboards/idobao/montex/v1rgb/rules.mk [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = no            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = WS2812
```
Comment (drashna) [inline keyboards/idobao/montex/v1rgb/rules.mk:19]:
```suggestion
RGB_MATRIX_DRIVER = WS2812

LAYOUTS = numpad_6x5
```

File: keyboards/idobao/montex/v1rgb/v1rgb.h [hunk 10]
```diff
@@ -0,0 +1,24 @@
+// Copyright 2022 peepeetee (@peepeetee)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "quantum.h"
+
+#define ___ KC_NO
+
+#define LAYOUT( \
```
Comment (drashna) [inline keyboards/idobao/montex/v1rgb/v1rgb.h:10]:
```suggestion
#define LAYOUT_numpad_6x5( \
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/idobao/montex/v1rgb/keymaps/via/config.h (hunk 5 above)
Comment (vinorodrigues) [reply keyboards/idobao/montex/v1rgb/keymaps/via/config.h:8]: not sure i understand - i'm not posting to develop, this is a new kb

**Review by vinorodrigues — APPROVED** [review]
fixes done.  leaving VIA keymap `VIA_QMK_RGBLIGHT_ENABLE` in as it will be needed for current VIA PR https://github.com/the-via/keyboards/pull/1250

File: keyboards/idobao/montex/v2/config.h (hunk 1 above)
Comment (vinorodrigues) [reply keyboards/idobao/montex/v2/config.h]: product ID changed to `0x0127` + new VIA json pending at the-via

File: keyboards/idobao/montex/v1rgb/keymaps/via/config.h (hunk 5 above)
Comment (vinorodrigues) [reply keyboards/idobao/montex/v1rgb/keymaps/via/config.h]: nm - i saw it in https://github.com/qmk/qmk_firmware/blob/e7931289918221081cbe2a7ea5df27a5d86324db/quantum/via.c#L38
deleting file instead

**Review by vinorodrigues — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Pull request merged**